common/achievement/index.ts

```diff
@@ -82,8 +82,7 @@ async function checkUserAchievement<ID extends ActionID>(userAchievement: UserAc
         const evaluationResultValue =
             typeof evaluationResult.resultObject[dataAggregationKey] === 'number' ? Number(evaluationResult.resultObject[dataAggregationKey]) : null;
         const awardedAchievement = await rewardUser(evaluationResultValue, userAchievement, event);
-        const userAchievementContext: UserAchievementContext = {};
-        await createAchievement(awardedAchievement.template, userAchievement.userId, userAchievementContext);
+        await createAchievement(awardedAchievement.template, userAchievement.userId, event.context);
```

Comment:
the only problem I see with using the same context as in the events is that the action `student_joined_match_meeting` / `pupil_joined_match_meeting` triggers more than one achievement. So if we store `match_all/1` and the join meeting action is also used for a specific achievement - for example "Regelmäßiges Lernen", where we have to check the lecture for just one match - than the buckets will not work anymore.

Reply:
so maybe we need a own action for global_match / global_subcourse achievements? Would this solve this?

Reply:
so what I mean is to store more achievement_events, that are more specific for each metric.

Reply:
<img src="https://media0.giphy.com/media/py2UYwTIX5SXm/giphy.gif"/>